setup.cfg

```diff
@@ -55,7 +55,85 @@ test =
     hypothesis>=5.5.3
     pytest>=6.0
     pytest-xdist>=1.31
-
+# optional extras for recommended dependencies
+# see: doc/source/getting_started/install.rst
+recommended =
+    bottleneck>=1.3.2
+    numexpr>=2.7.1
+computation =
+    numba>=0.53.0
+    scipy >=1.7.1
+    xarray>=0.19.0
+s3 =
+    boto3>=1.22.7
+    s3fs>=0.4.0
+gcp =
+    gcsfs>=2021.05.0
+    pandas-gbq>=0.15.0
+azure =
+    adlfs >=0.6.0
```

Comment:
As noted in a prior PR, we don't tests `azure` yet so I would not be confident including this as a category